File: sdk/fsharp/src/HiTechCloudSDK/Vcloudstack.fs
namespace HiTechCloud.SDK

/// vCloudStack Public Cloud: rescue, unrescue, console, usage
type VCloudStackResource(httpClient: HttpClient) =

    /// POST /api/service/{id}/vms/{vmid}/rescue - Reboot VM into rescue mode
    member this.RescueVm() =
        async {
            return! httpClient.postAsync("/api/service/{id}/vms/{vmid}/rescue")
        }

    /// POST /api/service/{id}/vms/{vmid}/unrescue - Disable rescue mode
    member this.UnrescueVm() =
        async {
            return! httpClient.postAsync("/api/service/{id}/vms/{vmid}/unrescue")
        }

    /// GET /api/service/{id}/vms/{vmid}/console - Get VM console
    member this.GetConsole() =
        async {
            return! httpClient.getAsync("/api/service/{id}/vms/{vmid}/console")
        }

    /// GET /api/service/{id}/vms/{vmid}/usage - Get VM usage details
    member this.GetUsage() =
        async {
            return! httpClient.getAsync("/api/service/{id}/vms/{vmid}/usage")
        }

File: sdk/clojure/src/hitechcloud/sdk/vcloudstack.clj
(ns hitechcloud.sdk.vcloudstack
  (:require [hitechcloud.sdk.http-client :as http]))

;; vCloudStack Public Cloud: rescue, unrescue, console, usage

(defn rescue_vm "POST /api/service/{id}/vms/{vmid}/rescue - Reboot VM into rescue mode" [client & [params]] (http-post client "/api/service/{id}/vms/{vmid}/rescue" params))

(defn unrescue_vm "POST /api/service/{id}/vms/{vmid}/unrescue - Disable rescue mode" [client & [params]] (http-post client "/api/service/{id}/vms/{vmid}/unrescue" params))

(defn get_console "GET /api/service/{id}/vms/{vmid}/console - Get VM console" [client & [params]] (http-get client "/api/service/{id}/vms/{vmid}/console" params))

(defn get_usage "GET /api/service/{id}/vms/{vmid}/usage - Get VM usage details" [client & [params]] (http-get client "/api/service/{id}/vms/{vmid}/usage" params))

